### ISSUE TITLE
Validate OCI repo names are valid

### ISSUE THREAD

**Issue by mattmoor** [sigstore/policy-controller#335]
**Description**

The validating webhook for CIPs happily accepts this:
```yaml
      source:
      - oci: gcr.io/mattmooor-chainguard/*
        signaturePullSecrets:
        - name: regcreds
```

... which is compiled into the configmap, and crashes the webhook when it tries to parse the containing policy

**Version**

v0.4

cc @hectorj2f @vaikas FYI

**Comment by hectorj2f** [comment]
Yes, we need to validate that repository name 😥 !

**Comment by hectorj2f** [comment]
Closing this issue! It has been fixed in #337
